source/common/router/config_impl.h

```diff
@@ -750,26 +697,63 @@ class RouteEntryImplBase : public RouteEntryAndRoute,
     return vhost_->virtualClusterFromEntries(headers);
   }
   std::chrono::milliseconds timeout() const override { return timeout_; }
+  bool usingNewTimeouts() const override { return using_new_timeouts_; }
+
+  // `OptionalTimeouts` manages various `optional` values. We pack them in a
+  // separate data structure for memory efficiency -- avoiding overhead of
+  // `absl::optional` per variable, and avoiding overhead of storing unset
+  // timeouts.
+  enum class OptionalTimeoutNames {
+    IdleTimeout = 0,
+    MaxStreamDuration,
+    GrpcTimeoutHeaderMax,
+    GrpcTimeoutHeaderOffset,
+    MaxGrpcTimeout,
+    GrpcTimeoutOffset
+  };
+  using OptionalTimeouts = PackedStruct<std::chrono::milliseconds, 6, OptionalTimeoutNames>;
+
   absl::optional<std::chrono::milliseconds> idleTimeout() const override {
-    return optional_timeouts_ != nullptr ? optional_timeouts_->idleTimeout() : absl::nullopt;
+    const auto timeout = optional_timeouts_.get<OptionalTimeoutNames::IdleTimeout>();
```

Comment:
Please remove duplication of these 6 accessors with a helper function (which may need to be a template function)

Reply:
Done